--- a/net/minecraft/world/level/entity/LevelEntityGetter.java
+++ b/net/minecraft/world/level/entity/LevelEntityGetter.java
@@ -17,6 +17,7 @@
     <U extends T> void get(EntityTypeTest<T, U> filter, Consumer<U> action);
 
     void get(AABB box, Consumer<T> action);
+    void get(AABB box, Consumer<T> action, boolean isContainerSearch); // Paper
 
     <U extends T> void get(EntityTypeTest<T, U> filter, AABB box, Consumer<U> action);
 }
