--- a/net/minecraft/world/entity/monster/RangedAttackMob.java
+++ b/net/minecraft/world/entity/monster/RangedAttackMob.java
@@ -3,5 +3,8 @@
 import net.minecraft.world.entity.LivingEntity;
 
 public interface RangedAttackMob {
-    void performRangedAttack(LivingEntity target, float pullProgress);
+    void performRangedAttack(LivingEntity target, float pullProgress); @Deprecated default void rangedAttack(LivingEntity entityliving, float f) { performRangedAttack(entityliving, f); } // Paper - OBFHELPER
+
+    // - see EntitySkeletonAbstract melee goal
+    void setAggressive(boolean flag); default void setChargingAttack(boolean charging) { setAggressive(charging); }; // Paper
 }
